From 95de7b79ed12daece43e2dfc64eb3232e1752c2b Mon Sep 17 00:00:00 2001
From: Nick Mathewson <nickm@torproject.org>
Date: Tue, 6 Feb 2024 08:16:25 -0500
Subject: [PATCH] Fix compilation with musl.

Don't try to name the type of `RLIMIT_FSIZE` in our tests:
its type is either `c_int` or `__rlimit_resource_t` depending on
the libc instance.

Closes #1264.
---
 crates/tor-hsservice/src/replay.rs | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/crates/tor-hsservice/src/replay.rs b/crates/tor-hsservice/src/replay.rs
index c82bf1ab7..ec0bb2771 100644
--- a/crates/tor-hsservice/src/replay.rs
+++ b/crates/tor-hsservice/src/replay.rs
@@ -526,15 +526,15 @@ mod test {
 
         fn set_ulimit(size: usize) {
             unsafe {
-                const RLIM: libc::__rlimit_resource_t = libc::RLIMIT_FSIZE;
+                use libc::RLIMIT_FSIZE;
                 let mut rlim = libc::rlimit {
                     rlim_cur: 0,
                     rlim_max: 0,
                 };
-                let r = libc::getrlimit(RLIM, (&mut rlim) as _);
+                let r = libc::getrlimit(RLIMIT_FSIZE, (&mut rlim) as _);
                 assert_eq!(r, 0);
                 rlim.rlim_cur = size.try_into().unwrap();
-                let r = libc::setrlimit(RLIM, (&rlim) as _);
+                let r = libc::setrlimit(RLIMIT_FSIZE, (&rlim) as _);
                 assert_eq!(r, 0);
             }
         }
-- 
2.43.0

